Frends.Community.Azure.Blob.Tests/DeleteTest.cs

```diff
@@ -9,72 +9,122 @@ namespace Frends.Community.Azure.Blob.Tests
             var input = new DeleteBlobProperties
             {
                 BlobName = Guid.NewGuid().ToString()
             };
             var connection = new BlobConnectionProperties
             {
+                ConnectionMethod = ConnectionMethod.ConnectionString,
                 ConnectionString = _connectionString,
-                ContainerName = _containerName
+                ContainerName = containerName
             };
-            var container = Utils.GetBlobContainer(_connectionString, _containerName);
+            var container = Utils.GetBlobContainer(_connectionString, containerName);
             await container.CreateIfNotExistsAsync();
 
             var result = await DeleteTask.DeleteBlobAsync(input, connection, new CancellationToken());
 
             Assert.IsTrue(result.Success, "DeleteBlob should've returned true when trying to delete non existing blob");
+            await DeleteContainer(containerName);
         }
 
         [TestMethod]
         public async Task DeleteBlobAsync_ShouldReturnTrueWithNonexistingContainer()
         {
+            var containerName = "test2";
+
             var input = new DeleteBlobProperties
             {
                 BlobName = Guid.NewGuid().ToString()
             };
             var options = new BlobConnectionProperties
             {
+                ConnectionMethod = ConnectionMethod.ConnectionString,
                 ConnectionString = _connectionString,
-                ContainerName = Guid.NewGuid().ToString()
+                ContainerName = containerName
             };
 
             var result = await DeleteTask.DeleteBlobAsync(input, options, new CancellationToken());
 
             Assert.IsTrue(result.Success,
                 "DeleteBlob should've returned true when trying to delete blob in non existing container");
+            await DeleteContainer(containerName);
+        }
+
+        [TestMethod]
+        public async Task DeleteBlobAsync_AccessTokenAuthenticationTest()
+        {
+            var containerName = "test3";
+            var input = new DeleteBlobProperties
+            {
+                BlobName = Guid.NewGuid().ToString()
+            };
+            var connection = new BlobConnectionProperties
+            {
+                ConnectionMethod = ConnectionMethod.AccessToken,
+                ContainerName = containerName,
+                StorageAccountName = _storageAccount,
+                ApplicationID = _appID,
+                TenantID = _tenantID,
+                ClientSecret = _clientSecret
+            };
+            var container = Utils.GetBlobContainer(_connectionString, containerName);
+            await container.CreateIfNotExistsAsync();
+
+            var result = await DeleteTask.DeleteBlobAsync(input, connection, new CancellationToken());
+
+            Assert.IsTrue(result.Success, "DeleteBlob should've returned true when trying to delete non existing blob");
+            await DeleteContainer(containerName);
```

Comment:
Is there any simple way to actually make sure that container was deleted or is this out of scope for this test? I mean for all we know the DeleteBlobAsync can just read "return true" and this test will pass :)

Reply:
I guess this comment above applies to some other unit tests as well.

Reply:
That is not a test, it's a cleaner. There was an issue with cleanup where tests would randomly fail due to containers being deleted when they shouldn't be, so I decided to implement the cleaner as a helper method to which you can specify which container should be deleted.